workers/task-driver/src/tasks/pay_offline_fee.rs

```diff
@@ -175,15 +179,14 @@ impl Task for PayOfflineFeeTask {
             .ok_or_else(|| PayOfflineFeeTaskError::State(ERR_WALLET_MISSING.to_string()))?;
 
         // Construct the new wallet
-        let (note, new_wallet) = Self::get_wallet_and_note(
-            descriptor.is_protocol_fee,
-            &descriptor.balance_mint,
-            &old_wallet,
-        )?;
+        let (note, new_wallet) = Self::get_wallet_and_note(&descriptor, &old_wallet)?;
+        if descriptor.amount != note.amount {
```

Comment:
hmm, let's say we have fee payment tasks enqueued, and a match (for the fee assets) preempts the queue. then this error will be tripped. this isn't a huge deal since the FE can retry the fee payment task, but it doesn't seem like it should cause an error. is it too overbearing to have historical task descriptors be calculated from tasks as opposed to task descriptors?

Reply:
well the case you described is quite the edge case and i think it's okay if that fails; note that a settle match won't be started if this task exists. 

It would end up being quite the refactor, getting all this information where it needs to be. And I don't love the idea of expanding the scope of this PR to cover a rare edge case